src/main/java/at/hannibal2/skyhanni/data/HotmData.kt

```diff
@@ -0,0 +1,672 @@
+            storage?.perks?.computeIfAbsent(this.name) { HotmTree.HotmPerk() }?.level = value
+        }
+
+    private fun blueEgg() = if (this != PEAK_OF_THE_MOUNTAIN && maxLevel != 1 && HotmAPI.isBlueEggActive) 1 else 0
+
+    var enabled: Boolean
+        get() = storage?.perks?.get(this.name)?.enabled ?: false
+        private set(value) {
+            storage?.perks?.computeIfAbsent(this.name) { HotmTree.HotmPerk() }?.enabled = value
+        }
+
+    var isUnlocked: Boolean
+        get() = storage?.perks?.get(this.name)?.isUnlocked ?: false
+        private set(value) {
+            storage?.perks?.computeIfAbsent(this.name) { HotmTree.HotmPerk() }?.isUnlocked = value
+        }
+
+    var slot: Slot? = null
+        private set
+
+    fun getLevelUpCost() = costFun(rawLevel)
+
+    fun getReward() = rewardFun(activeLevel)
+
+    companion object {
+
+        val storage get() = ProfileStorageData.profileSpecific?.mining?.hotmTree
+
+        val abilities =
+            listOf(PICKOBULUS, MINING_SPEED_BOOST, VEIN_SEEKER, MANIAC_MINER, HAZARDOUS_MINER, GEMSTONE_INFUSION)
+
+        private val inventoryPattern by repoGroup.pattern(
+            "inventory", "Heart of the Mountain"
+        )
+
+        private val levelPattern by repoGroup.pattern(
+            "perk.level", "§(?<color>.)Level (?<level>\\d+).*"
+        )
+
+        private val notUnlockedPattern by repoGroup.pattern(
+            "perk.notunlocked", "(§.)*Requires.*|.*Mountain!|(§.)*Click to unlock!|"
+        )
+
+        private val enabledPattern by repoGroup.pattern(
+            "perk.enable", "§a§lENABLED|(§.)*SELECTED"
+        )
+        private val disabledPattern by repoGroup.pattern(
+            "perk.disabled", "§c§lDISABLED|§7§eClick to select!"
+        ) // unused for now since the assumption is when enabled isn't found it is disabled, but the value might be useful in the future or for debugging
+
+        private val resetChatPattern by repoGroup.pattern(
+            "reset.chat", "§aReset your §r§5Heart of the Mountain§r§a! Your Perks and Abilities have been reset."
+        )
+
+        private val heartItemPattern by repoGroup.pattern(
+            "inventory.heart", "§5Heart of the Mountain"
+        )
+        private val resetItemPattern by repoGroup.pattern(
+            "inventory.reset", "§cReset Heart of the Mountain"
+        )
+
+        private val heartTokensPattern by repoGroup.pattern(
+            "inventory.heart.token", "§7Token of the Mountain: §5(?<token>\\d+)"
+        )
+
+        private val resetTokensPattern by repoGroup.pattern(
+            "inventory.reset.token", "\\s+§8- §5(?<token>\\d+) Token of the Mountain"
+        )
+
+        private val skymallPattern by repoGroup.pattern(
+            "skymall",
+            "(?:§eNew buff§r§r§r: §r§f|§8 ■ §7)(?<perk>.*)"
+        )
+
+        private val mayhemChatPattern by repoGroup.pattern(
+            "mayhem",
+            "§b§lMAYHEM! §r§7(?<perk>.*)"
+        )
```

Comment:
Should all be formatted like other patterns in the mod, and the empty line between the patterns is not necessary but can stay if you want

private val pattern by group.pattern(
	key,
	pattern
)